Rubberduck.Parsing/VBA/ParseCoordinator.cs

```diff
@@ -86,7 +97,35 @@ private void ReparseRequested(object sender, EventArgs e)
                 token = _currentCancellationTokenSource.Token;
             }
 
+            BeginParse(sender,token);
+        }
+
+        public void SuspendRequested(object sender, RubberduckStatusSuspendParserEventArgs e)
+        {
+            lock (_cancellationSyncObject)
+            {
+                if (State.Status != ParserState.Ready)
+                {
+                    throw new InvalidOperationException(
+                        "Cannot suspend the parser while it is running. Either cancel or wait for Ready status");
+                }
+
+                State.SetStatusAndFireStateChanged(e.Requestor, ParserState.Busy, CancellationToken.None);
+                e.BusyAction.Invoke();
+                if (_queuedRequestors.TryPop(out var lastRequestor))
```

Comment:
how can we guarantee that we're `Pop`-ing the last requestor here?  
And is the specific requestor even relevant? Could we instead maintain the equivalent of an `AtomicInteger` and an `AtomicBoolean` for "a parse-request was queued"?

Reply:
That was also more what I was thinking about. This can be done with the static methods on the `Interlocked` class.

Reply:
Just to double-check --- this variable might need to be renamed - this is now a `ConcurrentStack`, and as such, the `TryPop` is supposed to be thread-safe. Do we still need to use `Interlocked`? 

Regarding the requestor - I was following the pattern that the `OnParseRequested` uses; which simply pass among the requestor, and due to the fact that it may cancel a previous parser, the parse that completes is associated with the last requestor. Whether we need to know about prior requestors -- I don't know yet if there will be diagnostic value in having that information.